bundle/src/main/java/com/adobe/acs/commons/errorpagehandler/impl/ErrorPageHandlerImpl.java

```diff
@@ -514,18 +514,16 @@ private Resource findFirstRealParentOrSelf(SlingHttpServletRequest request, Reso
         }
 
         // If the resource exists, then use it!
-        if (resource != null && !ResourceUtil.isNonExistingResource(resource)) {
+        if (!ResourceUtil.isNonExistingResource(resource)) {
             log.debug("Found real aggregate resource at [ {} }", resource.getPath());
             return resource;
         }
 
         // Quick check for the Parent; Handles common case of deactivated pages
-        if (resource != null) {
-            final Resource parent = resource.getParent();
-            if (parent != null) {
-                log.debug("Found real aggregate resource via getParent() at [ {} ]", parent.getPath());
-                return parent;
-            }
+        final Resource parent = resource.getParent();
+        if (parent != null) {
+            log.debug("Found real aggregate resource via getParent() at [ {} ]", parent.getPath());
+            return parent;
```

Comment:
If Resource is not null, don't think parent of that resource could be null. isn't?